File: core/commands/registry/irules/psm__http__enable.py
# Enriched from F5 iRules reference documentation.
"""PSM::HTTP::enable -- To enable PSM for HTTP traffic."""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/PSM__HTTP__enable.html"


@register
class PsmHttpEnableCommand(CommandDef):
    name = "PSM::HTTP::enable"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="PSM::HTTP::enable",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="To enable PSM for HTTP traffic.",
                synopsis=("PSM::HTTP::enable",),
                snippet="To enable PSM for HTTP traffic",
                source=_SOURCE,
                examples=(
                    "when HTTP_REQUEST {\n"
                    "    PSM::HTTP::disable\n"
                    '    if { [HTTP::uri] starts_with "/enforce" } {\n'
                    "        PSM::HTTP::enable\n"
                    "    }\n"
                    "}"
                ),
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="PSM::HTTP::enable",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(
                profiles=frozenset({"HTTP"}), also_in=frozenset({"CLIENT_ACCEPTED"})
            ),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.NETWORK_IO,
                    writes=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/psm__http__enable.rs
//! `PSM::HTTP::enable` iRules command.
use crate::prelude::*;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "PSM::HTTP::enable",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
hover: Some(HoverSnippet {
            summary: "To enable PSM for HTTP traffic.",
            synopsis: &["PSM::HTTP::enable"],
            snippet: "To enable PSM for HTTP traffic",
            source: "https://clouddocs.f5.com/api/irules/PSM__HTTP__enable.html",
            examples: "when HTTP_REQUEST {\n    PSM::HTTP::disable\n    if { [HTTP::uri] starts_with \"/enforce\" } {\n        PSM::HTTP::enable\n    }\n}",
            return_value: "",
        }),
        event_requires: Some(EventRequires {
            client_side: false,
            server_side: false,
            transport: None,
            profiles: &["HTTP"],
            also_in: &["CLIENT_ACCEPTED"],
            init_only: false,
            flow: false,
            capability: None,
        }),
        forms: &[
            FormSpec { kind: FormKind::Default, synopsis: "PSM::HTTP::enable" },
        ],
        side_effects: &[
            SideEffect {
                target: SideEffectTarget::NetworkIo,
                reads: false,
                writes: true,
                connection_side: ConnectionSide::Both,
            },
        ],
        ..CommandSpec::DEFAULT
    }
}
